kinto/plugins/history/views.py

```diff
@@ -57,3 +61,83 @@ def _extract_filters(self):
             filters_str_id.append(filt)
 
         return filters_str_id
+
+
+version_view = Service(name='version_view',
+                       description='Handle retrieving object from the past.',
+                       path='/buckets/{bucket_id}{subpath:.*}/version/{last_modified:[0-9]{13}}')
+
+
+@version_view.get(permission=NO_PERMISSION_REQUIRED)
+def get_version_view(request):
+    last_modified = int(request.matchdict['last_modified'])
+    bucket_id = request.matchdict['bucket_id']
+    resource_path = '/buckets/{}{}'.format(request.matchdict['bucket_id'],
+                                           request.matchdict['subpath'])
+
+    bucket_uri = '/buckets/{}'.format(bucket_id)
+
+    is_collection = any([resource_path.endswith('/{}'.format(collection_type))
+                         for collection_type in ['collections', 'groups', 'records']])
+
+    # Handle collections
+    if is_collection:
+        return handle_version_on_collections(request, last_modified, bucket_uri)
+
+    # Handle records
+    return handle_version_on_records(request, last_modified, bucket_uri)
+
+
+def get_resource_name(request):
+    subpath = '/{}'.format(request.matchdict['subpath'])
+    resources = {
+        '/collections': "collection",
+        '/groups': 'group',
+        '/records': 'record'
+    }
```

Comment:
Good point! Please add some comments to the code, the conversations on github don't last